internal/types/array.go

```diff
@@ -111,3 +111,38 @@ func (a *Array) Append(values ...any) error {
 func (a *Array) RemoveByPath(path Path) {
 	removeByPath(a, path)
 }
+
+// Contains checks if the Array contains the given value.
+func (a *Array) Contains(value any) bool {
+	for _, v := range a.s {
+		if v == value {
+			return true
+		}
+	}
+	return false
+}
+
+// ContainsAll checks if the Array contains all the given values of the Array b.
+// Currently, this algorithm is O(n^2) without any performance tuning.
+// Possible improvement: to sorts both arrays before comparing.
+// If sorting is implemented, calling of the Contains function will be O(n*log(n)) instead of O(n^2).
+func (a *Array) ContainsAll(b *Array) bool {
+	if a.Len() < b.Len() {
+		// TODO: How to validate that all the values in b are unique?
```

Comment:
A controversial example:
`a`: `{1, 2}`
`b`: `{1, 1, 1}`

I believe, in this case, `ContainsAll` should return true. But for that, we either need to remove this `if a.Len() < b.Len()` condition or make `b` unique. 

@AlekSi what do you think?

Reply:
> A controversial example: `a`: `{1, 2}` `b`: `{1, 1, 1}`
> 
> I believe, in this case, `ContainsAll` should return `true`. 

What MongoDB international test gives? There are some things that are non-usual in Mongo and for better compatibility and faster migration to the FerretDB it has to be the same.

Reply:
I removed this condition to simplify the case.